action.yml

```diff
@@ -7,14 +7,20 @@ inputs:
     description: 'Personal access token with pull permissions on repository'
     required: true
 
+outputs:
+  errors:
+    description: 'Number of non conventional commit messages'
+    value: ${{ steps.run-script.outputs.errors }}
+
 runs:
   using: "composite"
   steps:
     - uses: actions/checkout@v4.1.1
       with:
         fetch-depth: 0
         token: ${{ inputs.token }}
-    - run: >
-        test "$(git log --grep='^\w+(\((\w|-)+\))?!?: .+$' --invert-grep -E --pretty=format:%s | awk 'END {print NR}')" -eq "0"
+    - id: run-script
+      run: >
+        [[ $(./script.sh) -eq "0" ]]
```

Comment:
The conditional expression is not needed: just running the script have the same result, as the exit status of the step is set to be the same exit status of the script. Still, I don't see an improvement in having a separate file for the script for such a simple action.

Side note: here you don't need pattern matching (i.e. `[[`) too, so `test` or `[` will suffice.

Reply:
Thank you, I'll use the `test` notation